ecs-cli/modules/cli/compose/project/project_parseV3.go

```diff
@@ -51,24 +74,92 @@ func (p *ecsProject) parseV3() (*[]containerconfig.ContainerConfig, error) {
+	//TODO: Add Healthcheck to ContainerConfig
+	c := &containerconfig.ContainerConfig{
+		CapAdd:  serviceConfig.CapAdd,
+		CapDrop: serviceConfig.CapDrop,
+		// Devices:               serviceConfig.Devices
+		DockerSecurityOptions: serviceConfig.SecurityOpt,
+		Name:             serviceConfig.Name,
+		Image:            serviceConfig.Image,
+		Hostname:         serviceConfig.Hostname,
+		Links:            serviceConfig.Links,
+		Privileged:       serviceConfig.Privileged,
+		ReadOnly:         serviceConfig.ReadOnly,
+		Command:          serviceConfig.Command,
+		User:             serviceConfig.User,
+		WorkingDirectory: serviceConfig.WorkingDir,
+	}
+
+	if serviceConfig.DNS != nil {
+		c.DNSServers = serviceConfig.DNS
+	}
+	if serviceConfig.DNSSearch != nil {
+		c.DNSSearchDomains = serviceConfig.DNSSearch
+	}
+	if serviceConfig.Entrypoint != nil {
+		c.Entrypoint = serviceConfig.Entrypoint
+	}
+	if serviceConfig.ExtraHosts != nil {
+		extraHosts, err := utils.ConvertToExtraHosts(serviceConfig.ExtraHosts)
 		if err != nil {
 			return nil, err
 		}
-		conConfigs = append(conConfigs, *cCon)
+		c.ExtraHosts = extraHosts
 	}
+	if serviceConfig.Tmpfs != nil {
+		tmpfs := yaml.Stringorslice(serviceConfig.Tmpfs)
+		ecsTmpfs, err := utils.ConvertToTmpfs(tmpfs)
+		if err != nil {
+			return nil, err
+		}
+		c.Tmpfs = ecsTmpfs
+	}
+	if serviceConfig.Logging != nil {
+		logConfig := ecs.LogConfiguration{}
+		logConfig.SetLogDriver(serviceConfig.Logging.Driver)
 
-	return &conConfigs, nil
-}
+		optionsMap := makePointerMapForStringMap(serviceConfig.Logging.Options)
+		logConfig.SetOptions(optionsMap)
+		c.LogConfiguration = &logConfig
+	}
+	if serviceConfig.Labels != nil {
+		labelsMap := makePointerMapForStringMap(serviceConfig.Labels)
+		c.DockerLabels = labelsMap
+	}
+	if len(serviceConfig.Ports) > 0 {
+		var portMappings []*ecs.PortMapping
+		for _, portConfig := range serviceConfig.Ports {
+			mapping := convertPortConfigToECSMapping(portConfig)
+			portMappings = append(portMappings, mapping)
+		}
+		c.PortMappings = portMappings
+	}
+	// TODO: reconcile with top-level Volumes key
+	if len(serviceConfig.Volumes) > 0 {
+		mountPoints := []*ecs.MountPoint{}
 
-func convertToContainerConfig(serviceConfig types.ServiceConfig) (*containerconfig.ContainerConfig, error) {
-	//TODO: fully convert ServiceConfig to ContainerConfig
-	c := &containerconfig.ContainerConfig{
-		Name: serviceConfig.Name,
+		for _, volConfig := range serviceConfig.Volumes {
+			if volConfig.Type == "volume" {
+				mp := &ecs.MountPoint{
+					ContainerPath: &volConfig.Target,
+					SourceVolume:  &volConfig.Source,
+					ReadOnly:      &volConfig.ReadOnly,
+				}
+				mountPoints = append(mountPoints, mp)
+			} else {
+				log.Warnf("Unsupported mount type found: %s", volConfig.Type)
+			}
+		}
+		c.MountPoints = mountPoints
 	}
+
+	// TODO: add Environtment, EnvFile to ContainerConfig
+	// TODO: log out unsupported fields
```

Comment:
Also noting for posterity: Other TODOs include CPU/Memory for a container. Also, I noticed that `ShmSize` does appear to be supported in docker compose v3, but does not appear to be in the docker/cli `ServiceConfig` type. Is there some other way to  specify shared memory size?

Reply:
Since CPU/Memory is defined on the `deploy` key in v3, we won't be setting it here (but we will should log it out with other unsupported fields that are found). Tentatively, we'll be relying on values in ecs-params.yml that we'll add in convert_task_definition